docs/node-operators/migration-guides/upgrade-node.md

```diff
@@ -1,40 +1,168 @@
+    }
+} else {
+    if startHeight == 0 {
+         // Replay the chain or state-sync if enabled. Up to tendermint to decide.
+    } else {
+	    // Wait for state-sync to offer expected snapshot for height
+    }
+}
+```
+
+#### `TrustedProxies`
+
+- `config file`: data-node
+- `description`: The parameter defines trusted proxies which can override the real user IP address with`X-Real-IP` or `X-Forward-For` headers. If the proxy address IP is not in the list of `TrustedProxies`, Vega ignores the above headers. You must repeat this parameter for the `gRPC API server` and the `Gateway server`.
+- `kind`: new parameter
+
+```toml title="YOUR_VEGA_HOME/config/data-node/config.toml"
+[API]
+  [API.RateLimit]
+    TrustedProxies = ["127.0.0.1"]
+
+[Gateway]
+  [Gateway.RateLimit]
+    TrustedProxies = ["127.0.0.1"]
+```
+
+#### `NetworkHistory.GarbageCollectionInterval`
+
+- `config file`: data-node
+- `description`: The interval at which garbage collection should be run.
+- `kind`: new parameter
+
+```toml title="YOUR_VEGA_HOME/config/data-node/config.toml"
+[NetworkHistory]
+    GarbageCollectionInterval = "24h0m0s"
+```
+
+#### `SQLStore.VerboseMigration`
+
+- `config file`: data-node
+- `description`: Enable verbose logging of SQL migrations. Logs shows only once when Vega executes the migration. We recommend enabling it in case of any issues.
+- `kind`: new parameter
+
+```toml title="YOUR_VEGA_HOME/config/data-node/config.toml"
+[SQLStore]
+  VerboseMigration = true
+```
 
-When you connect the data node to the validator node, and migration takes very long time, your validator **will not start** before the data node. This may impact your performance score and increase penalties.
+#### `NetworkHistory.Initialise.TimeOut`
 
-Changing the retention policy of your data node will minimise your validator node's downtime.
+- `config file`: data-node
+- `description`: We recommend you update it to `96 hours` due to changes in the data node initialization mechanism. It now downloads and loads all the segments into the database by default. This process may take a day or more.
+- `kind`: parameter change
 
-Execute this step at least a few hours before the mainnet upgrade.
+```toml
+[NetworkHistory]
+  [NetworkHistory.Initialise]
+    TimeOut = "96h0m0s"
+```
 
-Follow the below instructions:
+#### `NetworkHistory.Initialise.MinimumBlockCount`
 
-1. Update the `<vega_home>/config/data-node/config.toml`: `SQLStore.RetentionPeriod = "standard"`
-2. Restart your data node
+- `config file`: data-node
+- `description`: The network history initialization process has been improved. Now the process is automated. The datanode should download all the history segments and put it into the database by default now.
+- `kind`: parameter change
 
-Note: You are strongly recommended to only run a data node with a non-validator node, not a validator node.
+```toml
+[NetworkHistory]
+  [NetworkHistory.Initialise]
+    MinimumBlockCount = -1
+```
 
```

Comment:
We have also updated the default value in vegavisor `config.toml`: `maxNumberOfFirstConnectionRetries`

Reply:
updated